__tests__/commands/run-integration.js

```diff
@@ -0,0 +1,57 @@
+/* @flow */
+
+import makeTemp from '../_temp.js';
+import * as fs from '../../src/util/fs.js';
+import {run as buildRun} from './_helpers.js';
+import {BufferReporter} from '../../src/reporters/index.js';
+import {run} from '../../src/cli/commands/run.js';
+
+const path = require('path');
+
+const fixturesLoc = path.join(__dirname, '..', 'fixtures', 'run');
+const runRun = buildRun.bind(null, BufferReporter, fixturesLoc, (args, flags, config, reporter): Promise<void> => {
+  return run(config, reporter, flags, args);
+});
+
+jasmine.DEFAULT_TIMEOUT_INTERVAL = 30000;
+
+test('fully quote and escape extra args to run', async () => {
+  const trickyStrings = ['$X', '%X%', '^', '!', '\\', '>', '<', '|', '&', "'", '"', '`', '  '];
+  const cases = [];
+  for (const a of trickyStrings) {
+    for (const b of trickyStrings) {
+      cases.push(a + b + b + b);
+      if (a !== b) {
+        cases.push(a + a + b + b);
+        cases.push(a + b + a + b);
+        cases.push(a + b + b + a);
+        cases.push(a + a + a + b);
+        cases.push(a + a + b + a);
+        cases.push(a + b + a + a);
+      }
+    }
+  }
+  cases.sort(() => Math.random() - 0.5);
```

Comment:
It's bad if true randomness is required, but it's good enough for a quick scramble (would you prefer I implement/`require` a Knuth shuffle just for this test?). I'm randomizing because I'm not actually testing every combination; I didn't want to take too long on this test but I also wanted to get a decent sampling of the space. Basically it's a poor man's property-based test. If you think it's a good idea, I'd be happy to pull in jsverify or jssmartcheck and write a real property-based test. Alternatively, if this is a library, the heavy property checking can go there and yarn can test a much smaller sampling of cases.